view/ViewableData.php

```diff
@@ -127,6 +134,41 @@ public function __set($property, $value) {
 			$this->setField($property, $value);
 		}
 	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function __call($method, $arguments) {
+		$this->checkAndRedefineMethods();
+		return parent::__call($method, $arguments);
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function hasMethod($method) {
+		$this->checkAndRedefineMethods();
```

Comment:
+1. If you have it all over the place (especially in __call()) then its going to redefine all methods every time something on the failover object changes.